server/src/main/java/server/haengdong/application/BillService.java

```diff
@@ -0,0 +1,188 @@
+import server.haengdong.application.request.BillAppRequest;
+import server.haengdong.application.request.BillDetailUpdateAppRequest;
+import server.haengdong.application.request.BillDetailsUpdateAppRequest;
+import server.haengdong.application.request.BillUpdateAppRequest;
+import server.haengdong.application.response.BillDetailsAppResponse;
+import server.haengdong.application.response.LastBillMemberAppResponse;
+import server.haengdong.application.response.StepAppResponse;
+import server.haengdong.domain.action.Bill;
+import server.haengdong.domain.action.BillDetail;
+import server.haengdong.domain.action.BillRepository;
+import server.haengdong.domain.action.Member;
+import server.haengdong.domain.action.MemberRepository;
+import server.haengdong.domain.event.Event;
+import server.haengdong.domain.event.EventRepository;
+import server.haengdong.exception.HaengdongErrorCode;
+import server.haengdong.exception.HaengdongException;
+
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+@Service
+public class BillService {
+
+    private final BillRepository billRepository;
+    private final EventRepository eventRepository;
+    private final MemberRepository memberRepository;
+
+    @Transactional
+    public void saveBill(String eventToken, BillAppRequest request) {
+        Event event = getEvent(eventToken);
+        List<Long> memberIds = request.memberIds();
+        List<Member> members = memberIds.stream()
+                .map(this::findMember)
+                .toList();
+
+        Bill bill = request.toBill(event, members);
+        billRepository.save(bill);
+    }
+
+    private Member findMember(Long memberId) {
+        return memberRepository.findById(memberId)
+                .orElseThrow(() -> new HaengdongException(HaengdongErrorCode.MEMBER_NOT_FOUND));
+    }
+
+    public List<StepAppResponse> findSteps(String token) {
+        Event event = getEvent(token);
+        List<Bill> bills = billRepository.findByEvent(event);
+
+        return createStepAppResponses(bills);
+    }
+
+    private List<StepAppResponse> createStepAppResponses(List<Bill> bills) {
+        return divideByMembers(bills).stream()
+                .map(StepAppResponse::of)
+                .toList();
+    }
+
+    private static List<List<Bill>> divideByMembers(List<Bill> bills) {
+        List<List<Bill>> split = new ArrayList<>();
+        for (Bill bill : bills) {
+            if (split.isEmpty()) {
+                List<Bill> temp = new ArrayList<>();
+                temp.add(bill);
+                split.add(temp);
+                continue;
+            }
+            List<Bill> bills1 = split.get(split.size() - 1);
+            Bill find = bills1.get(0);
+            if (find.isSameMembers(bill)) {
+                bills1.add(bill);
+            } else {
+                List<Bill> temp = new ArrayList<>();
+                temp.add(bill);
+                split.add(temp);
+            }
+        }
+        return split;
+    }
```

Comment:
```suggestion
    private List<List<Bill>> divideByMembers(List<Bill> bills) {
        List<List<Bill>> billGroup = new ArrayList<>();
        List<Bill> currentGroup = null;

        for (Bill bill : bills) {
            if (currentGroup == null || !currentGroup.get(0).isSameMembers(bill)) {
                currentGroup = new ArrayList<>();
                billGroup.add(currentGroup);
            }
            currentGroup.add(bill);
        }
        
        return billGroup;
    }
```
조금 단순화 해봤습니다. 그리고 static 메서드일 필요가 없네요.

Reply:
이 로직을 Steps, Step 객체를 만들어 넣어도 좋겠네요.